internal/services/cdn/cdn_endpoint_custom_domain_resource_test.go

```diff
@@ -80,6 +111,76 @@ func TestAccCdnEndpointCustomDomain_requiresImport(t *testing.T) {
 	})
 }
 
+func TestAccCdnEndpointCustomDomain_httpsCdn(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_cdn_endpoint_custom_domain", "test")
+
+	preCheck(t)
+
+	r := NewCdnEndpointCustomDomainResource(os.Getenv("ARM_TEST_DNS_ZONE_RESOURCE_GROUP_NAME"), os.Getenv("ARM_TEST_DNS_ZONE_NAME"))
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.httpsCdn(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccCdnEndpointCustomDomain_httpsCdnUpdate(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_cdn_endpoint_custom_domain", "test")
+
+	preCheck(t)
+
+	r := NewCdnEndpointCustomDomainResource(os.Getenv("ARM_TEST_DNS_ZONE_RESOURCE_GROUP_NAME"), os.Getenv("ARM_TEST_DNS_ZONE_NAME"))
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.httpsCdn(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccCdnEndpointCustomDomain_httpsUserManagedBasic(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_cdn_endpoint_custom_domain", "test")
+
+	preCheckUserManagedCertificate(t)
+
+	r := NewCdnEndpointCustomDomainResource(os.Getenv("ARM_TEST_DNS_ZONE_RESOURCE_GROUP_NAME"), os.Getenv("ARM_TEST_DNS_ZONE_NAME")).
+		WithCertificate(os.Getenv("ARM_TEST_DNS_CERTIFICATE")).
+		WithSubDomain(os.Getenv("ARM_TEST_DNS_SUBDOMAIN_NAME"))
```

Comment:
(same here) can we move this onto the resource here?

```suggestion
	r := NewCdnEndpointCustomDomainResource(os.Getenv("ARM_TEST_DNS_ZONE_RESOURCE_GROUP_NAME"), os.Getenv("ARM_TEST_DNS_ZONE_NAME"))
	r.Certificate = os.Getenv("ARM_TEST_DNS_CERTIFICATE")
	r.SubDomain = os.Getenv("ARM_TEST_DNS_SUBDOMAIN_NAME")
	r.preCheckUserManagedCertificate(t)
```